src/components/context/DataProvider.tsx

```diff
@@ -200,6 +202,15 @@ const DataProvider = ({ children }: Props): JSX.Element => {
     }
   }, [explorerData, view, actions, explorerIsError]);
 
+  // create a color map for item actions to be unified through the whole app
+  const actionsColorMap = useMemo(() => {
+    const allActions = [...new Set(actions.map((a) => a.type))];
+    return allActions.reduce((acc: { [key: string]: string }, curr, index) => {
+      acc[curr] = COLORS[index];
+      return acc;
+    }, {});
+  }, [actions]);
```

Comment:
Let's take the opportunity to complete "ActionTriggers" from sdk and fill the `COLORS` array with as many colors.
Doing so will allow you to create a constant Map `ActionTriggerToColor` (no need for dynamic function) `triggerString -> color`. Let's also define a default color in case the trigger enum is not complete.